s3transfer/crt.py

```diff
@@ -599,11 +646,27 @@ def result(self, timeout=None):
             self._crt_future.result(timeout)
         except KeyboardInterrupt:
             self.cancel()
+            self._crt_future.result(timeout)
             raise
+        except Exception as e:
+            self.handle_exception(e)
         finally:
             if self._s3_request:
                 self._s3_request = None
-            self._crt_future.result(timeout)
+
+    def handle_exception(self, exc):
+        translated_exc = None
+        if self._exception_translator:
+            try:
+                translated_exc = self._exception_translator(exc)
+            except Exception:
+                # Bail out if we hit an issue translating
```

Comment:
We should probably log if there was an exception and that we are ignoring it in favor the translated exception